jets/f/ut_find.c

```diff
@@ -13,18 +13,74 @@
              u3_noun lon,
              u3_noun gil);
 
+/*  `u3qfu_felt_arm` is a helper function for
+ *  u3qfu_felt.  It handles the case in which the
+ *  opal is for an arm, and hence which needs a 
+ *  list of `hold`s.  These will be converted to
+ *  a `fork` type.
+*/
   static u3_noun
-  u3qfu_felt(u3_noun van,
-             u3_noun sut,
-             u3_noun lap)
+  u3qfu_felt_arm(u3_noun lis)
   {
-    u3_noun von = u3i_molt(u3k(van), u3x_sam, u3k(sut), 0);
-    u3_noun gat = u3j_cook("u3qfu_felt-felt", von, "felt");
+      if ( lis == u3_nul ) {
+          return u3_nul;
+      } else {
+          
+          u3_noun i_lis = u3h(lis);
+          u3_noun t_lis = u3t(lis);
+          u3_noun typ = u3h(i_lis);
+
+          u3_noun p_typ, q_typ, pq_typ, qq_typ, rq_typ;
+
+          if ( (c3n == u3du(typ)) ||
+               (u3h(typ) != c3__core) ) {
+              return u3m_error("felt-core");
+          } else if
+              ( (c3n == u3r_cell(u3t(typ), &p_typ, &q_typ)) ||
+                (c3n == u3r_trel(q_typ, &pq_typ, &qq_typ, &rq_typ)) )
+          {
+              return u3m_bail(c3__fail);
```

Comment:
you want %exit here, not %fail. %exit is normal crash (!!), %fail is like... we ran out of memory or something, a non-deterministic, resource-exhaustion-caused crash.